dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-raft/src/main/java/org/apache/dolphinscheduler/plugin/registry/raft/RaftRegistry.java

```diff
@@ -0,0 +1,188 @@
+                .withCommonNodeOptions(nodeOptions)
+                .config();
+        final RheaKVStoreOptions opts = RheaKVStoreOptionsConfigured.newConfigured()
+                .withClusterName(properties.getClusterName())
+                .withUseParallelCompress(true)
+                .withInitialServerList(properties.getServerAddressList())
+                .withStoreEngineOptions(storeOpts)
+                .withPlacementDriverOptions(pdOpts)
+                .config();
+        this.kvStore = new DefaultRheaKVStore();
+        this.kvStore.init(opts);
+        log.info("kvStore started...");
+        this.ephemeralNodeManager = new EphemeralNodeManager(properties, kvStore);
+    }
+
+    @PostConstruct
+    public void start() {
+        ephemeralNodeManager.start();
+    }
+
+    @Override
+    public boolean subscribe(String path, SubscribeListener listener) {
+        return ephemeralNodeManager.addSubscribeListener(path, listener);
+    }
+
+    @Override
+    public void unsubscribe(String path) {
+        ephemeralNodeManager.removeSubscribeListener(path);
+    }
+
+    @Override
+    public void addConnectionStateListener(ConnectionListener listener) {
+        ephemeralNodeManager.addConnectionListener(listener);
+    }
+
+    @Override
+    public String get(String key) {
+        return readUtf8(kvStore.bGet(key));
+    }
+
+    @Override
+    public void put(String key, String value, boolean deleteOnDisconnect) {
+        if (StringUtils.isBlank(value)) {
+            return;
+        }
+        readUtf8(kvStore.bGetAndPut(key, writeUtf8(value)));
+        ephemeralNodeManager.putHandler(key, value);
+    }
+
+    @Override
+    public void delete(String key) {
+        kvStore.bDelete(key);
+        final DistributedLock<byte[]> distributedLock = distributedLockMap.get(key);
+        if (distributedLock != null) {
+            distributedLock.unlock();
+        }
+        distributedLockMap.remove(key);
+        ephemeralNodeManager.deleteHandler(key);
+
+    }
+
+    @Override
+    public Collection<String> children(String key) {
+        final String result = readUtf8(kvStore.bGet(key));
+        if (StringUtils.isEmpty(result)) {
+            return new ArrayList<>();
+        }
+        final List<String> children = JSONUtils.toList(result, String.class);
+        children.sort(Comparator.reverseOrder());
+        return children;
+    }
+
+    @Override
+    public boolean exists(String key) {
+        return kvStore.bContainsKey(key);
+    }
+
+    @Override
+    public boolean acquireLock(String key) {
+        final DistributedLock<byte[]> distributedLock = kvStore.getDistributedLock(key, properties.getDistributedLockTimeout().toMillis(), TimeUnit.MILLISECONDS);
+        final boolean lock = distributedLock.tryLock();
+        if (lock) {
+            distributedLockMap.put(key, distributedLock);
+        }
+        return lock;
```

Comment:
Is it  better to catch `Unable to get lock exception`  and throw RegistryException manually？

Reply:
not need